plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -95,14 +95,24 @@ public function save_to_session( $customer ) {
 				 * @param WC_Customer $customer The customer object.
 				 */
 				$allowed_keys  = apply_filters( 'woocommerce_customer_allowed_session_meta_keys', array(), $customer );
-				$session_value = wp_json_encode(
-					array_filter(
-						$customer->get_meta_data(),
-						function( $meta_data ) use ( $allowed_keys ) {
-							return in_array( $meta_data->key, $allowed_keys, true );
-						}
+				$session_value = maybe_serialize(
+					array_map(
+						function ( $meta_data ) {
+							// Data comes to us a WC_Meta_Data, we cast it to an array to ensure it is serializable.
```

Comment:
```suggestion
							// Data comes to us as WC_Meta_Data, we cast it to an array to ensure it is serializable.
```

It seems there's a typo: _"comes to us a WC_Meta_Data"_ vs. _"comes to us as WC_Meta_Data"_.